drop-in/src/main/java/com/adyen/checkout/dropin/internal/ui/ManageFavoritesScreen.kt

```diff
@@ -64,59 +68,97 @@ private fun ManageFavoritesContent(
                 favorites = viewState.others,
-                onItemClick = {},
+                onItemClick = { selectedItem = it },
+            )
+        }
+
+        selectedItem?.let { item ->
+            ConfirmationDialog(
+                // TODO - Pass item title as parameter after we resolve parameterized strings
+                confirmationText = resolveString(CheckoutLocalizationKey.DROP_IN_MANAGE_FAVORITES_REMOVE_CONFIRMATION),
+                onConfirmationClick = {
+                    onRemoveItem(item)
+                    selectedItem = null
+                },
+                cancellationText = resolveString(CheckoutLocalizationKey.GENERAL_CANCEL),
+                onDismissRequest = { selectedItem = null },
             )
         }
     }
 }
 
-@Composable
-private fun Section(
-    title: String,
+private fun LazyListScope.section(
+    title: CheckoutLocalizationKey,
     favorites: List<FavoriteListItem>,
     onItemClick: (FavoriteListItem) -> Unit,
 ) {
-    favorites.forEachIndexed { index, item ->
-        if (index == 0) {
+    if (favorites.isNotEmpty()) {
+        item(key = title) {
             SubHeadlineEmphasized(
-                text = title,
-                modifier = Modifier.padding(horizontal = Dimensions.Large, vertical = Dimensions.Small),
+                text = resolveString(title),
+                modifier = Modifier
+                    .padding(horizontal = Dimensions.Large, vertical = Dimensions.Small)
+                    .animateItem(),
             )
         }
+    }
 
-        ListItem(
-            leadingIcon = {
-                CheckoutNetworkLogo(
-                    txVariant = item.icon,
-                    modifier = Modifier.size(Dimensions.LogoSize.medium),
-                )
-            },
-            title = item.title,
-            subtitle = item.subtitle,
-            trailingIcon = { Body(text = "Remove", color = CheckoutThemeProvider.colors.destructive) },
-            onClick = { onItemClick(item) },
-            modifier = Modifier.padding(horizontal = Dimensions.ExtraSmall),
-        )
+    favorites.forEach { item ->
+        item(key = item.id) {
+            FavoriteListItem(
+                item = item,
+                onClick = onItemClick,
+                modifier = Modifier
+                    .padding(horizontal = Dimensions.ExtraSmall)
+                    .animateItem(),
+            )
+        }
     }
 }
 
+@Composable
+private fun FavoriteListItem(
+    item: FavoriteListItem,
+    onClick: (FavoriteListItem) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    ListItem(
+        leadingIcon = {
+            CheckoutNetworkLogo(
+                txVariant = item.icon,
+                modifier = Modifier.size(Dimensions.LogoSize.medium),
+            )
+        },
+        title = item.title,
+        subtitle = item.subtitle,
+        trailingIcon = {
```

Comment:
Now that we're passing a text composable, `trailingIcon` name doesn't apply anymore. We could use `trailingContent` as name? Also for consistency we could change the `leadingIcon` as well.

Reply:
`trailingContent` makes a lot of sense. Not sure if we should rename `leadingIcon` as IMO this should really only be used for icons and not for text or anything like that

Reply:
Fair, let's leave `leadingIcon` as is to at least indicate that it's for icons only.